src/Compilers/CSharp/Test/Syntax/Parsing/MemberDeclarationParsingTests.cs

```diff
@@ -458,5 +458,287 @@ public void TrashAfterDeclaration()
             }
             EOF();
         }
+
+        [Fact]
+        [WorkItem(11959, "https://github.com/dotnet/roslyn/issues/11959")]
+        public void GenericAsyncTask_01()
+        {
+            foreach (var options in new[] { TestOptions.Script, TestOptions.Regular })
+            {
+                UsingDeclaration("async Task<SomeNamespace.SomeType Method();", options: options,
+                    // (1,1): error CS1073: Unexpected token '('
+                    // async Task<SomeNamespace.SomeType Method();
+                    Diagnostic(ErrorCode.ERR_UnexpectedToken, "async Task<SomeNamespace.SomeType Method").WithArguments("(").WithLocation(1, 1),
+                    // (1,35): error CS1003: Syntax error, ',' expected
+                    // async Task<SomeNamespace.SomeType Method();
+                    Diagnostic(ErrorCode.ERR_SyntaxError, "Method").WithArguments(",", "").WithLocation(1, 35),
+                    // (1,41): error CS1003: Syntax error, '>' expected
+                    // async Task<SomeNamespace.SomeType Method();
+                    Diagnostic(ErrorCode.ERR_SyntaxError, "(").WithArguments(">", "(").WithLocation(1, 41)
```

Comment:
why are there so many diagnostics?  shouldn't we just want a diagnostic for a single missing `>`?

Reply:
Because it considers "async Task<SomeNamespace.SomeType Method();"
as 
"async Task<SomeNamespace.SomeType,Method>();"

So, Method goes as the second type argument and it could not find the member identified after that. Then, it could not understand why '()' are provided.

We discussed this with @AlekseyTs. It seems there it would be useful to have re-considering this parsing but not for now.

---
In reply to: [252524863](https://github.com/dotnet/roslyn/pull/32983#discussion_r252524863) [](ancestors = 252524863)

Reply:
> We discussed this with @AlekseyTs. It seems there it would be useful to have re-considering this parsing but not for now.

Pity.  That parsing is truly terrible :)   I'll put it on my backburner to see if there's a way to improve things.